force-app/main/default/lwc/donationHistoryTable/donationHistoryTable.js

```diff
@@ -33,17 +33,30 @@ export default class DonationHistoryTable extends LightningElement {
         if (data) {
             this.paymentMethodLabel = data.fields[PAYMENT_FIELD.fieldApiName].label
         };
-        this.columns = [
-            { label: RD2_ScheduleVisualizerColumnDate, fieldName: 'closeDate', type: 'date-local', typeAttributes:{
-                year: "numeric",
-                month: "numeric",
-                day: "numeric",
-            },
-            cellAttributes: { alignment: 'right' }},
-            { label: donationHistoryDonorLabel, fieldName: 'name', type: 'text' },
-            { label: commonAmount, fieldName: 'amount', type: 'currency', },
-            { label: this.paymentMethodLabel, fieldName: 'paymentMethod', type: 'text', },
-        ];
+        arePaymentsEnabled({contactId: this.contactId}).then( (paymentsAreEnabled) => {
+            if(paymentsAreEnabled){
+                this.columns = [
+                { label: RD2_ScheduleVisualizerColumnDate, fieldName: 'closeDate', type: 'date-local', typeAttributes:{
+                    year: "numeric",
+                    month: "numeric",
+                    day: "numeric",
+                },
+                cellAttributes: { alignment: 'right' }},
+                { label: donationHistoryDonorLabel, fieldName: 'name', type: 'text' },
+                { label: commonAmount, fieldName: 'amount', type: 'currency', },
+                { label: this.paymentMethodLabel, fieldName: 'paymentMethod', type: 'text', },];
+            }else{
```

Comment:
} else {

Reply:
Cant we changed this for something like:
this.columns = [
                { label: RD2_ScheduleVisualizerColumnDate, fieldName: 'closeDate', type: 'date-local', typeAttributes:{
                    year: "numeric",
                    month: "numeric",
                    day: "numeric",
                },
                cellAttributes: { alignment: 'right' }},
                { label: donationHistoryDonorLabel, fieldName: 'name', type: 'text' },
                { label: commonAmount, fieldName: 'amount', type: 'currency', }]
if (paymentsAreEnabled) {
    this.columns.push(...)
}

Im not a JS expert but I think something like that is doable